src/ManagedServices/ManagedServices/Commands/NewAzureRmManagedServicesDefinition.cs

```diff
@@ -0,0 +1,145 @@
+        public string ManagedByTenantId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managedby Principal Identifier.")]
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The Managedby Principal Identifier.")]
+        public string PrincipalId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managed Service Providers's Role Identifier.")]
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The Managed Service Providers's Role Identifier.")]
+        public string RoleDefinitionId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The description of the Registration Definition.")]
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = false, HelpMessage = "The description of the Registration Definition.")]
+        public string Description { get; set; }
+
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The name of the plan.")]
+        [ValidateNotNullOrEmpty]
+        public string PlanName { get; set; }
+
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The name of the Publisher.")]
+        [ValidateNotNullOrEmpty]
+        public string PlanPublisher { get; set; }
+
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The name of the Product.")]
+        [ValidateNotNullOrEmpty]
+        public string PlanProduct { get; set; }
+
+        [Parameter(ParameterSetName = ByPlanParameterSet, Mandatory = true, HelpMessage = "The version number of the plan.")]
+        [ValidateNotNullOrEmpty]
+        public string PlanVersion { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public Guid RegistrationDefinitionId { get; set; } = default(Guid);
+
+        public override void ExecuteCmdlet()
+        {
+            if (!this.ManagedByTenantId.IsGuid())
+            {
+                throw new ApplicationException("ManagedByTenantId must be a valid GUID.");
+            }
+
+            if (!this.PrincipalId.IsGuid())
+            {
+                throw new ApplicationException("PrincipalId must be a valid GUID.");
+            }
+
+            if (!this.RoleDefinitionId.IsGuid())
+            {
+                throw new ApplicationException("RoleDefinitionId must be a valid GUID.");
+            }
+
+            if (this.RegistrationDefinitionId == default(Guid))
+            {
+                this.RegistrationDefinitionId = Guid.NewGuid();
+            }
+
+            ConfirmAction(MyInvocation.InvocationName,
```

Comment:
This message should be specific about what is being created at what scope - people use these messages to debug scripts, for example.  - at least the name of the registration definition, and where it is being created